# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit/CacheIRCompiler.cpp
# Commit: e9db0721d85d
# Full Hash: e9db0721d85d94e0514b3e83a4e87bdc1198d89a
# Author: Matthew Gaudet <mgaudet@mozilla.com>
# Date: 2024-02-01 05:30:14
# Description:
#   Bug 1877684 - Add support for boolean constants in CacheIRRegisterAllocator::useRegister r=anba
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D200206
# ==============================================================================

diff -r fedb6cd9dddf -r e9db0721d85d js/src/jit/CacheIRCompiler.cpp
--- a/js/src/jit/CacheIRCompiler.cpp	Wed Jan 31 20:10:04 2024 +0000
+++ b/js/src/jit/CacheIRCompiler.cpp	Wed Jan 31 20:10:20 2024 +0000
@@ -356,6 +356,8 @@
         masm.movePtr(ImmGCPtr(v.toSymbol()), reg);
       } else if (v.isBigInt()) {
         masm.movePtr(ImmGCPtr(v.toBigInt()), reg);
+      } else if (v.isBoolean()) {
+        masm.movePtr(ImmWord(v.toBoolean() ? 1 : 0), reg);
       } else {
         MOZ_CRASH("Unexpected Value");
       }
